### PR TITLE
More work getting reduce ready

### PR DESCRIPTION
**Pull request by mariari**

### CODE REVIEW

File: src/Juvix/Backends/LLVM/Net/EAC.hs [hunk 1]
```diff
@@ -169,9 +257,23 @@ reduce = Codegen.defineFunction Type.void "reduce" args $
 
 -- this function work off the nodeType signature not Types.eac
 
+anihilateRewireAux ∷
```
Comment (cwgoes) [inline src/Juvix/Backends/LLVM/Net/EAC.hs]:
```suggestion
annihilateRewireAux ∷
```

File: src/Juvix/Backends/LLVM/Net/EAC.hs [hunk 2]
```diff
@@ -169,9 +257,23 @@ reduce = Codegen.defineFunction Type.void "reduce" args $
...
+  ( HasThrow "err" Codegen.Errors m,
+    HasState "blocks" (Map.HashMap Name.Name Codegen.BlockState) m,
+    HasState "count" Word m,
+    HasState "currentBlock" Name.Name m,
+    HasState "symtab" Codegen.SymbolTable m
+  ) ⇒
+  [Operand.Operand] →
+  m ()
+anihilateRewireAux args = do
```
Comment (cwgoes) [inline src/Juvix/Backends/LLVM/Net/EAC.hs]:
```suggestion
annihilateRewireAux args = do
```

File: src/Juvix/Backends/LLVM/Net/EAC.hs [hunk 3]
```diff
@@ -169,9 +257,23 @@ reduce = Codegen.defineFunction Type.void "reduce" args $
...
+  anihilate ← Codegen.externf "anihilate_rewire_aux"
```
Comment (cwgoes) [inline src/Juvix/Backends/LLVM/Net/EAC.hs]:
```suggestion
  annihilate ← Codegen.externf "annihilate_rewire_aux"
```

File: src/Juvix/Backends/LLVM/Net/EAC.hs [hunk 4]
```diff
@@ -169,9 +257,23 @@ reduce = Codegen.defineFunction Type.void "reduce" args $
...
+  _ ← Codegen.call Type.void anihilate (Codegen.emptyArgs args)
```
Comment (cwgoes) [inline src/Juvix/Backends/LLVM/Net/EAC.hs]:
```suggestion
  _ ← Codegen.call Type.void annihilate (Codegen.emptyArgs args)
```

File: src/Juvix/Backends/LLVM/Net/EAC.hs [hunk 5]
```diff
@@ -169,9 +257,23 @@ reduce = Codegen.defineFunction Type.void "reduce" args $
...
+  pure ()
+
 -- mimic rules from the interpreter
 -- This rule applies to Application ↔ Lambda
-anihilateRewireAux ∷
+anihilateRewireAux' ∷
```
Comment (cwgoes) [inline src/Juvix/Backends/LLVM/Net/EAC.hs]:
```suggestion
annihilateRewireAux' ∷
```

File: src/Juvix/Backends/LLVM/Net/EAC.hs [hunk 6]
```diff
@@ -184,7 +286,7 @@ anihilateRewireAux ∷
     HasState "varTab" Codegen.VariantToType m
   ) ⇒
   m Operand.Operand
-anihilateRewireAux = Codegen.defineFunction Type.void "anihilate_rewire_aux" args $
+anihilateRewireAux' = Codegen.defineFunction Type.void "anihilate_rewire_aux" args $
```
Comment (cwgoes) [inline src/Juvix/Backends/LLVM/Net/EAC.hs]:
```suggestion
annihilateRewireAux' = Codegen.defineFunction Type.void "anihilate_rewire_aux" args $
```

**Comment by mariari** [comment]
Typos fixed

**Pull request merged**